AAP 1505--Update docs for using custom SSL certificates in OCP (#554)
RedHatInsights/red-hat-ansible-automation-platform-documentation: downstream/modules/platform/proc-controller-image-pull-policy.adoc

[id="proc_controller-image-pull-policy_{context}"]

> = controller-image-pull-policy

Reviewer: Is the title of this procedure intentional? Per modularization standards, we should begin procedure modules with a gerund. For example, we could say "Configuring your automation controller image pull policy". I'm just guessing. I have no idea what the task is here, but it should say that clearly in the title instead of using the file name as the title.

> 

Reviewer: I would suggest adding an abstract of some type to this procedure because the "why" for this topic is not clear. See the current section or sub-chapter "3.2.1 Configure your automation controller operator route options" (although this title displays another inconsistency in our docs with regards to modularization rules; it should be "Configuring...").  In another example, sub-chapter "3.2 Installing the automation controller operator" doesn't necessarily need an abstract because you understand why it's necessary, if that makes sense.

Author: I would dare to guess that anyone setting up an image pull policy understands it is the current image they are pulling, none, always, or IfNotPresent, which means the image gets pulled if it is not already on the node. These are Kubernetes rules, not Red Hat's, so most people familiar with Kubernetes, and this I gather from conversations I have had, would know these rules and set them accordingly.

> . Under *Image Pull Policy*, click on the radio button to select *Always*, *Never*, or *IfNotPresent*.

Reviewer: ```suggestion
. Under *Image Pull Policy*, select one of the following: 
..*Always* 
..*Never*
..*IfNotPresent*
```

> . Define the *Image Pull Secrets*

Reviewer: ```suggestion
. Complete the following steps to define the *Image Pull Secrets*.
```
Added punctuation. The rest is just a suggestion. If you think this provides too much info, please feel free to ignore.

> . Define the *Image Pull Secrets*

Reviewer: ```suggestion
. To display options under *Image Pull Secrets*, click the arrow to the right.
```
This is a suggestion to capture exactly what the user will be doing. You can also find examples of this in the IBM Style Guide in the [UI elements ](https://www.ibm.com/docs/en/ibm-style?topic=elements-ui)section (See Drop-down list in the table)

> .. Click on the angle bracket to show the *Add Image Pull Secret*.

Reviewer: ```suggestion
.. Click the angle bracket to show the *Add Image Pull Secret*.
```

> .. Click on the angle bracket to show the *Add Image Pull Secret*.

Author: Value is the only field there is. So that is still accurate. A good question on whether or not there are values, I have sent an inquiry to Christian, though I suspect that the answer is that it depends upon the user's hardware specs.

> .. Click on the angle bracket to show the *Add Image Pull Secret*.

Reviewer 2: The value is the name of the k8s secret that contains credentials for accessing a private image registry.  The user would know the name of the pull secret because they will have created it.  

![image](https://user-images.githubusercontent.com/11698892/193652814-79fe4c4a-f0d4-4e97-ac4c-5c6ddf10c136.png)

Note that this can be plural (Image Pull Secret (s)).  

Docs on creating image pull secrets:
* https://kubernetes.io/docs/tasks/configure-pod-container/pull-image-private-registry/
* https://docs.openshift.com/container-platform/4.10/openshift_images/managing_images/using-image-pull-secrets.html

> .. Click on the angle bracket to show the *Add Image Pull Secret*.

Reviewer 2: Also, this is optional.  If customers are only using images from the official redhat image registry [redhat.registry.io](http://redhat.registry.io/) (which is the default), then they do not need to create and specify an image pull secret.

> .. Click on the *Add Image Pull Secret* and enter the *Value* in the field.

Reviewer: ```suggestion
.. Click *Add Image Pull Secret* and enter a *Value* .
```
Another suggestion to make things concise. Just a question. Are there parameters for the value field? Will the customer know what the values should be? Just thinking through what a user would expect.

> .. Click on the *Add Image Pull Secret* and enter the *Value* in the field.

Reviewer: ```suggestion
.. Click on the *Add Image Pull Secret* and enter the *Value* in the field.
```
If you decide to accept my suggestion above, you really won't need this step since it will already be covered.

> . Define the *Web container resource requirements* by clicking on the angle brackets to display the *Web container resource requirements* fields.

Reviewer: Complete the following steps to define the *Web container resource requirements*:

> . Define the *Web container resource requirements* by clicking on the angle brackets to display the *Web container resource requirements* fields.

Reviewer: ```suggestion
. Complete the following steps to define the *Web container resource requirements*:
```
Just a suggestion.

> . Define the *Web container resource requirements* by clicking on the angle brackets to display the *Web container resource requirements* fields.

Reviewer: ```suggestion
. To display the *Web container resource requirements* fields, click the arrow on the right.
```
Just a suggestion to make the wording more concise and eliminate possibilities of confusion.

> .. Enter the necessary values for *Limits*, and *Requests*.

Reviewer: ```suggestion
.. Enter values in the *Limits*, and *Requests* fields.
```
Just a suggestion. Question: Are there defined values for these fields or will the users/customers know what the values should be? Are there criteria for what goes into these fields? Otherwise, this seems arbitrary.

> .. Enter the necessary values for *Limits*, and *Requests*.

Reviewer: ```suggestion
.. Under **Limits** and **Requests**, enter values for **CPU cores**, **Memory**, and **Storage**.
```

> .. Enter the necessary values for *Limits*, and *Requests*.

Reviewer: ```suggestion
.. Under *Limits*, and *Requests*, enter values for *CPU cores*, *Memory*, and *Storage*.
```
Just a suggestion to be specific and exact about the fields that users are required to enter.

Reviewer 2: Users are not _required_ to enter limits or requests here.  There are default values which will work in most cases. 

I am not sure if there is a way that we denote if a field is optional or not, but the only "required" field for a simple AutomationController deployment is the "name" field.

> . Define the *Task container resource requirements* by clicking on the angle bracket to display the resource requirements fields.

Reviewer: ```suggestion
. To display the *Task container resource requirements* options, click the arrow.
```
Suggestion again. See IBM style guide in the [UI elements ](https://www.ibm.com/docs/en/ibm-style?topic=elements-ui)section (specifically, Drop-down list in the table)

> .. Enter the necessary values for *Limits*, and *Requests*.

Reviewer: ```suggestion
.. Under *Limits*, and *Requests*, enter values for *CPU cores*, *Memory*, and *Storage*.
```
Again, just a suggestion for clarity, conciseness, and adherence to style guides.

> Define the *EE Control Plane container resource requirements* by clicking on the angle bracket to display the resource requirements fields.

Reviewer: ```suggestion
To display fields under the *EE Control Plane container resource requirements* drop-down list, click the arrow.
```
Suggestion based on styles and standards.

> .. Enter the necessary values for *Limits*, and *Requests*.

Reviewer: ```suggestion
.. Under *Limits*, and *Requests*, enter values for *CPU cores*, *Memory*, and *Storage*.
```

> Define the *PostgreSQL init container resource requirements (when using a managed service)* by clicking on the angle bracket to display the resource requirements fields.

Reviewer: ```suggestion
To display fields under the *PostgreSQL init container resource requirements (when using a managed service)* drop-down list, click the arrow.
```
Suggestion based on IBM'S UI elements topic (similar to previous steps).

> .. Enter the necessary values for *Limits*, and *Requests*.

Reviewer: ```suggestion
.. Under *Limits*, and *Requests*, enter values for *CPU cores*, *Memory*, and *Storage*.
```
Suggestion only.

> Define the *Redis container resource requirements* by clicking on the angle bracket to display the resource requirements fields.

Reviewer: ```suggestion
To display fields under the *Redis container resource requirements* drop-down list, click the right arrow.
```

> .. Enter the necessary values for *Limits*, and *Requests*.

Reviewer: Under *Limits*, and *Requests*, enter values for *CPU cores*, *Memory*, and *Storage*.

> .. Enter the necessary values for *Limits*, and *Requests*.

Reviewer: ```suggestion
.. Under *Limits*, and *Requests*, enter values for *CPU cores*, *Memory*, and *Storage*.
```

> . Define the *PostgreSQL container resource requirements (when using a managed instance)** by clicking on the angle bracket to display the resource requirements fields.

Reviewer: ```suggestion
. To display fields under the *PostgreSQL container resource requirements (when using a managed instance)** drop-down list, click the arrow.
```

> .. Enter the necessary values for *Limits*, and *Requests*.

Reviewer: ```suggestion
.. Under *Limits*, and *Requests*, enter values for *CPU cores*, *Memory*, and *Storage*.
```

> . Define the *PostgreSQL container storage requirements (when using a managed instance)* by clicking on the angle bracket to display the resource requirements fields.

Reviewer: ```suggestion
. To display the *PostgreSQL container storage requirements (when using a managed instance)* drop-down list, click the arrow.
```

> .. Enter the necessary values for *Limits*, and *Requests*.

Reviewer: ```suggestion
.. Under *Limits*, and *Requests*, enter values for *CPU cores*, *Memory*, and *Storage*.
```

> . Enter the number of *Replicas*.

Reviewer: I'm not necessarily suggesting a change here, but for consistency with other steps, you could change this to: 

Under *Replicas*, enter the number of instance replicas.

> . Under *Remove used secrets on instance removal* select either true or false by moving the slider. The default is false.

Reviewer: ```suggestion
. Under *Remove used secrets on instance removal*, move the slider to either true or false. The default is false.
```
Just a suggestion.

> . Under *Preload instance with data upon creation* select either true or false by moving the slider. The default is true. 

Reviewer: ```suggestion
. Under *Preload instance with data upon creation*, select *true* or *false*. The default is true. 
```
